podcasts/DownloadManager+URLSessionDelegate.swift

```diff
@@ -63,8 +68,16 @@ extension DownloadManager: URLSessionDelegate, URLSessionDownloadDelegate {
         switch error.code {
         case NSURLErrorCancelled:
             if !episode.downloadFailed() {
-                // already handled this error, since we failed the download ourselves
+                // we already handled this error, since we failed the download ourselves
+                let reason = error.userInfo[NSURLErrorBackgroundTaskCancelledReasonKey] as? Int
+                switch reason {
+                case NSURLErrorCancelledReasonUserForceQuitApplication, NSURLErrorCancelledReasonInsufficientSystemResources:
+                    DataManager.sharedManager.saveEpisode(downloadStatus: .queued, downloadTaskId: nil, episode: episode)
```

Comment:
When the system cancels a download task, which should reset it to `queued` so we don't fail to download it when the app restarts or a background refresh re-triggers downloads.